pkg/arvo/app/publish.hoon

```diff
@@ -1702,8 +1702,88 @@
     :~  [%give %fact [/primary]~ %publish-primary-delta !>(act)]
         [%give %fact [/publishtile]~ %json !>(jon)]
     ==
+  ::  %groupify
+  ::
+      %groupify
+    ?.  (team:title our.bol src.bol)
+      ~|("action not permitted" !!)
+    =/  book  (~(get by books) our.bol book.act)
+    ?~  book
+      ~|("nonexistent notebook: {<book.act>}" !!)
+    ::
+    =/  old-write      writers.u.book
+    =/  old-read  subscribers.u.book
+    ?>  ?=([%'~' ^] old-write)
+    =/  destroy-old-groups=(list card)
+      :~  (group-poke [%unbundle old-write])
+          (group-poke [%unbundle old-read])
+          (group-hook-poke [%remove old-write])
+          (group-hook-poke [%remove old-read])
+          (perm-group-hook-poke [%dissociate old-write (sy old-write ~)])
+          (perm-group-hook-poke [%dissociate old-read (sy old-read ~)])
+          (perm-hook-poke [%remove old-write])
+          (perm-hook-poke [%remove old-read])
+      ==
+    ::
+    ?~  target.act
+      :: create new group from subscribers
+      =.  writers.u.book      (slag 1 writers.u.book)
+      =.  subscribers.u.book  writers.u.book
+      =/  del=notebook-delta  [%edit-book our.bol book.act u.book]
+      :_  state(books (~(put by books) [our.bol book.act] u.book))
+      %+  weld  destroy-old-groups
+      ^-  (list card)
+      :~  [%give %fact [/notebook/[book.act]]~ %publish-notebook-delta !>(del)]
+          [%give %fact [/primary]~ %publish-primary-delta !>(del)]
+          %-  contact-view-create
+          :*  writers.u.book
+              (get-subscribers book.act)
+              title.u.book
+              description.u.book
+          ==
+      ==
+    ?>  ?=(^ u.target.act)
+    =.  writers.u.book  u.target.act
+    =.  subscribers.u.book  u.target.act
+    =/  group-host=@p  (slav %p i.u.target.act)
+    =/  scry-pax  :(weld /=group-store/(scot %da now.bol) u.target.act /noun)
+    =/  old-group=(set @p)  (need .^((unit (set @p)) %gx scry-pax))
+    =/  dif-peeps=(set @p)  (~(dif in (get-subscribers book.act)) old-group)
+    ?:  ?&  inclusive.act
+            =(group-host our.bol)
+        ==
+      :: add all subscribers to group
+      =/  del=notebook-delta  [%edit-book our.bol book.act u.book]
+      :_  state(books (~(put by books) [our.bol book.act] u.book))
+      %+  weld  destroy-old-groups
+      ^-  (list card)
+      :~  [%give %fact [/notebook/[book.act]]~ %publish-notebook-delta !>(del)]
+          [%give %fact [/primary]~ %publish-primary-delta !>(del)]
+          (group-poke [%add dif-peeps u.target.act])
+      ==
+    :: kick subscribers who are not already in group
+    =/  del=notebook-delta  [%edit-book our.bol book.act u.book]
+    :_  state(books (~(put by books) [our.bol book.act] u.book))
+    %+  weld  destroy-old-groups
+    ^-  (list card)
+    :+  [%give %fact [/notebook/[book.act]]~ %publish-notebook-delta !>(del)]
+      [%give %fact [/primary]~ %publish-primary-delta !>(del)]
```

Comment:
It looks like these lines are equivalent between the "add new subscribers to group" and "kick subscribers not in group" cases. If so, you could just move the `?:` into the tail of the `:+` here and cut down a bunch of LoC.